src/hotspot/share/oops/inlineKlass.cpp

```diff
@@ -447,8 +441,21 @@ void InlineKlass::initialize_calling_convention(TRAPS) {
   if (InlineTypeReturnedAsFields || InlineTypePassFieldsAsArgs) {
     ResourceMark rm;
     GrowableArray<SigEntry> sig_vk;
-    float max_offset = 0;
-    int nb_fields = collect_fields(&sig_vk, max_offset);
+    int nb_fields = collect_fields(&sig_vk);
+    if (*PrintInlineKlassFields != '\0') {
+      const char* class_name_str = _name->as_C_string();
+      if (StringUtils::class_list_match(PrintInlineKlassFields, class_name_str)) {
```

Comment:
Nice! I already added similar printing multiple times when debugging issues in this area. Not sure why I never integrated that.